nntrainer/layers/layer_node.cpp

```diff
@@ -184,75 +180,10 @@ LayerNode::LayerNode(std::unique_ptr<nntrainer::Layer> &&l) :
 }
 
 void LayerNode::setProperty(const std::vector<std::string> &properties) {
-  bool already_distributed =
-    !std::get<props::Distribute>(*layer_node_props).empty() &&
-    std::get<props::Distribute>(*layer_node_props).get();
   auto left_properties = loadProperties(properties, *layer_node_props);
-
-  /// note that setting distribute is only allowed for one time.
-  /// until we have layerNode::finalize and must not except timedist layer
-  if (getDistribute() && !already_distributed) {
-    auto &ac = nntrainer::AppContext::Global();
-    std::unique_ptr<nntrainer::Layer> dlayer =
-      ac.createObject<nntrainer::Layer>(TimeDistLayer::type);
-    if (dlayer.get() == nullptr)
-      throw std::invalid_argument("Error creating time distribution layer");
-    auto *time_dist_layer = dynamic_cast<TimeDistLayer *>(dlayer.get());
-    if (time_dist_layer == nullptr)
-      throw std::invalid_argument("Error casting to time distribution layer");
-    time_dist_layer->setDistLayer(std::move(layer));
-    layer = std::move(dlayer);
-  }
-
-  std::vector<std::string> remainder;
-  /// @todo: deprecate this in favor of loadProperties
-  for (unsigned int i = 0; i < left_properties.size(); ++i) {
-
-    std::string key;
-    std::string value;
-    std::stringstream ss;
-
-    if (getKeyValue(left_properties[i], key, value) != ML_ERROR_NONE) {
-      throw std::invalid_argument("Error parsing the property: " +
-                                  left_properties[i]);
-    }
-
-    if (value.empty()) {
-      ss << "value is empty: key: " << key << ", value: " << value;
-      throw std::invalid_argument(ss.str());
-    }
-
-    /// @note this calls derived setProperty if available
-    if (!setProperty(key, value)) {
-      remainder.push_back(left_properties[i]);
-    }
-  }
-
-  layer->setProperty(remainder);
-}
-
-bool LayerNode::setProperty(const std::string &key, const std::string &value) {
-  using PropertyType = nntrainer::Layer::PropertyType;
-
-  PropertyType type = static_cast<PropertyType>(parseLayerProperty(key));
-  switch (type) {
-  case PropertyType::activation: {
-    setActivation((ActivationType)parseType(value, TOKEN_ACTI));
-    if (getType() == ActivationLayer::type) {
-      ml_logi("Set property delegated to activation layer");
-      return false;
-    }
-    break;
-  }
-  case PropertyType::num_inputs: {
-    ml_logw("Deprecated property: %s", key.c_str());
-    break;
-  }
-  default:
-    return false;
-  }
-
-  return true;
+  left_properties =
+    loadProperties(left_properties, *layer_node_props_realization);
+  layer->setProperty(left_properties);
```

Comment:
AFAIK, TimeDistLayer is added when the property is set. Looks like this is deleted.

Reply:
Yes, if we set TimeDistLayer when setProperty,
setProperty({"timedist=true", "timedist=true"}) is a bug.

So it's moved to `finalize`